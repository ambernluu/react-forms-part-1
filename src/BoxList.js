import React, { useState } from 'react';
import NewBoxForm from './NewBoxForm';
import Box from './Box.js';

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);
    const addBox = (box) => {
        setBoxes(boxes => [...boxes, box])
    };
    const removeBox = id => {
        setBoxes(boxes => boxes.filter(box => box.id !== id));
    }
    const boxComponents = boxes.map(box => (
        <Box
            id={box.id}
            width={box.width}
            height={box.height}
            backgroundColor={box.backgroundColor}
            key={box.id}
            handleRemove={removeBox}
        />
    ));

    return (
        <div>
            <NewBoxForm addBox={addBox} />
            {boxComponents}
        </div>
    )

}

export default BoxList;