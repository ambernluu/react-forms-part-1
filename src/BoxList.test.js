import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing", () => {
    render(<BoxList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

function addBox(boxList, width = '150', height = '150', backgroundColor = 'cyan') {
    const widthInput = boxList.getByLabelText("Width (px)");
    const heightInput = boxList.getByLabelText('Height (px)');
    const backgroundInput = boxList.getByLabelText('Background Color');

    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    fireEvent.change(backgroundInput, { target: { value: backgroundColor } });
    const button = boxList.getByText('Create New Box!');
    fireEvent.click(button);

}

it("can add box", function () {
    const boxList = render(<BoxList />);

    expect(boxList.queryByText('Remove Box')).not.toBeInTheDocument();

    addBox(boxList);

    const removeButton = boxList.getByText('Remove Box');
    expect(removeButton).toBeInTheDocument();
    expect(removeButton.previousSibling).toHaveStyle(`
    width: 150px;
    height: 150px;
    background-color: cyan;
  `);

    expect(boxList.getAllByDisplayValue("")).toHaveLength(3);

});