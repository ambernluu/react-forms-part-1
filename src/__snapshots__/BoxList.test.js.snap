// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width (px) 
      </label>
      <input
        id="width"
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height (px) 
      </label>
      <input
        id="height"
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Background Color 
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Create New Box!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
